File: OOAD_HW1/Q4/cat.java
package Q4;

// Inherits from feline, which inherits from animal.
public class cat extends feline {
	
	// Cat's name
	private String name;
	
	// Default Constructor, don't use it.
	public cat() {
		super("None");
		this.name = "None";
	}
	
	// Overridden constructor for setting the cat's name, along with all super-class' name.
	public cat(String name) {
		super(name);
		this.name = name;
	}
	
	// Since cats cannot roar, override the makeNoise from feline class with "meow" instead of "roar"
	public void makeNoise() {
		System.out.println(this.name + " meows.");
	}
}

File: OOAD_HW2/Part A/Cat.cpp
// Cat Class

#include <iostream>
#include <string>
#include <cstdlib>
#include <iomanip>

//#include "Feline.cpp"

using namespace std;

class Cat : public Feline
{
    private:
        string name;
        string type;

    public:
        Cat(string aName, string aType);
        //~Cat();
        void makeNoise(void);
};

Cat::Cat(string aName, string aType): Feline(aName,aType)
{
    name = aName;
    type = aType;
}

void Cat::makeNoise(void)   // Cats cant roar. All other actions handled by super class.
{
    cout << name << ", the " << type << " meows." << endl;
}
